sdk/spring/spring-cloud-azure-core/src/main/java/com/azure/spring/core/resource/AzureStorageBlobProtocolResolver.java

```diff
@@ -87,10 +88,14 @@ public String getName() {
             ListBlobsOptions options = new ListBlobsOptions();
             options.setPrefix(itemPrefix);
             options.setDetails(RETRIEVE_NOTHING_DETAILS);
-            return getBlobServiceClient().getBlobContainerClient(name)
-                                    .listBlobs(options, null)
-                                    .stream()
-                                    .map(blob -> new StorageItem(name, blob.getName(), getStorageType()));
+            BlobContainerClient containerClient = getBlobServiceClient().getBlobContainerClient(name);
+            if (containerClient.exists()) {
```

Comment:
Consider catching and handling relevant storage code, sample here https://github.com/Azure/azure-sdk-for-java/blob/aff24131127ed85da1e76ed7cb4cccd9a25279c0/sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/APISpec.groovy#L128 .